server/core/src/main/java/com/redhat/thermostat/server/core/web/CustomResourceConfig.java

```diff
@@ -0,0 +1,24 @@
+package com.redhat.thermostat.server.core.web;
+
+import org.glassfish.jersey.server.ResourceConfig;
+import java.util.Map;
+import com.redhat.thermostat.server.core.internal.web.configuration.ServerConfiguration;
+import io.swagger.jaxrs.config.BeanConfig;
+
+public class CustomResourceConfig extends ResourceConfig {
+    public CustomResourceConfig(Map<String, String> serverConfig) {
+
+        if (serverConfig.containsKey(ServerConfiguration.SWAGGER_UI_ENABLED.toString()) &&
+                serverConfig.get(ServerConfiguration.SWAGGER_UI_ENABLED.toString()).equals("true")) {
+            BeanConfig beanConfig = new BeanConfig();
```

Comment:
It's not clear to me why this needs to be done inside the ResourceConfig class. From [1] it seems like it could go anywhere in the initialization of the Server. Would it make more sense to place this along with the lines:

+            resourceConfig.register(new ApiListingResource());
+            resourceConfig.register(new SwaggerSerializers());

in CoreServer::setupResourceHandler?


[1] https://github.com/swagger-api/swagger-core/wiki/Swagger-Core-RESTEasy-2.X-Project-Setup